### ISSUE TITLE
When using Azure SAML Login a "content type" header is sent with the GET Request

### ISSUE THREAD

**Issue by cheflA1** [brave/brave-browser#37003]
## Description 
When using Microsoft Azure SAML Login on web sites, brave (and other chrome based browsers) send a content type header with the GET request.
This gets blocked by WAFs, because a content type header implies that there is a body, which a GET request does not have.
We notices this first on 21.3.24.

## Steps to Reproduce

   1.  Use Azure SAML Login on a website

## Actual result:
This leads to traffic being blocked by WAFs, because of the faulty GET Request.

![image](https://github.com/brave/brave-browser/assets/30794959/e73c960d-0142-4cdd-89bb-85a44c54b815)

## Expected result:
No Content-Type Header in Get Request

## Reproduces how often: 
Everytime since yesterday.

## Brave version (brave://version info)
Brave | 1.64.109 Chromium: 123.0.6312.58 (Official Build) (64-bit)
-- | --
Revision | ded360c0dc8a4bd8aecb9cbe32a7aa4477d42246
OS | Windows 11 Version 23H2 (Build 22631.3155)
JavaScript | V8 12.3.219.10
User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36
Command Line | "C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe" --disable-domain-reliability --enable-dom-distiller --enable-distillability-service --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --flag-switches-begin --flag-switches-end --flag-switches-begin --flag-switches-end --component-updater=url-source=https://go-updater.brave.com/extensions,url-source=https://go-updater.brave.com/extensions
Executable Path | C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe
Profile Path | C:\Users\bpieron\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default
Active variations | AllowCertainClientHintsStudy:EnabledBraveAIChatEnabledStudy:EnabledBraveAdblockMobileNotificationsListDefault:EnabledBraveAutoTranslateStudy:AutoTranslateDisabledBraveDebounceStudy:EnabledBraveFeedUpdateStudy:EnabledBraveForgetFirstPartyStorage:EnabledBraveGoogleSignInPermissionStudy:DisabledBraveHorizontalTabsUpdateEnabledStudy:EnabledBraveHttpsByDefaultRolloutStudy:EnabledBraveP3AConstellationEnabled:EnabledBraveP3ATypicalJSONDeprecationEnabled:EnabledBraveRewardsAllowSelfCustodyProvidersStudy:EnabledBraveRewardsWebUiPanelStudy:EnabledBraveScreenFingerprintingBlockerStudy:EnabledBraveSearchAdStudy:EnabledClampPlatformVersionClientHint:DisabledCookieListDefaultStudy:EnabledCrossPlatformVPNStudy:EnabledDefault1pBlockingStudy:DisabledDisableReduceLanguage:EnabledEphemeralStorageStudy:EnabledPartitionedCookies:EnabledSpeedreader TTS:EnabledUserActivityStudy:TriggersWhatsNewStudy:Enabled

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? I don't know
- Can you reproduce this issue with the nightly channel? I don't know

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? Yes

## Miscellaneous Information:
We made an exception on our WAF, but we do not want to keep it this way of course.

**Comment by ShivanKaul** [comment]
Did you try reporting this to Chromium?

**Comment by ShivanKaul** [comment]
FWIW, I don't see this being outlawed by the RFC: https://www.rfc-editor.org/rfc/rfc9110#section-8.3. Though it is strange behaviour. Is this after a redirect?

**Comment by cheflA1** [comment]
> Did you try reporting this to Chromium?

not yet.

**Comment by cheflA1** [comment]
> FWIW, I don't see this being outlawed by the RFC: https://www.rfc-editor.org/rfc/rfc9110#section-8.3. Though it is strange behaviour. Is this after a redirect?

Yes. When clicking "Login with Azure" it does a redirect which works fine.

**Comment by ShivanKaul** [comment]
I would prefer to not differ from Chromium here since it's not a privacy issue, is fundamentally harmless and could cause webcompat problems down the line. If you could raise a Chromium bug and link it here we can track it: https://issues.chromium.org/new

**Comment by cheflA1** [comment]
> I would prefer to not differ from Chromium here since it's not a privacy issue, is fundamentally harmless and could cause webcompat problems down the line. If you could raise a Chromium bug and link it here we can track it: https://issues.chromium.org/new

I just reported it to Chromium.
Issue ID: 331406958
